src/System.Linq/src/System/Linq/Enumerable.SelectEnumerableIterator.cs

```diff
@@ -0,0 +1,64 @@
+﻿
+using System.Collections.Generic;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        internal class SelectEnumerableIterator<TSource, TResult> : SelectIterator<TResult>
+        {
+            private IEnumerable<TSource> _source;
+            private Func<TSource, TResult> _selector;
+            private IEnumerator<TSource> _enumerator;
+
+            public SelectEnumerableIterator(IEnumerable<TSource> source, Func<TSource, TResult> selector)
+            {
+                _source = source;
+                _selector = selector;
+            }
+
+            public override Iterator<TResult> Clone()
+            {
+                return new SelectEnumerableIterator<TSource, TResult>(_source, _selector);
+            }
+
+            public override void Dispose()
+            {
+                if (_enumerator is IDisposable) ((IDisposable)_enumerator).Dispose();
```

Comment:
Per the [style guidelines](https://github.com/dotnet/corefx/wiki/Coding-style), braces may be omitted but the block must be properly indented on its own line.